src/fs/shrinkwrap.ts

```diff
@@ -4,12 +4,30 @@ import {
   read as readYaml,
   write as writeYaml
 } from './yamlfs'
-import {ResolveResult} from '../resolve'
+import {PackageSpec, Resolution} from '../resolve'
 
 const shrinkwrapFilename = 'shrinkwrap.yaml'
 
 export type Shrinkwrap = {
-  [dependency: string]: ResolveResult
+  [dependency: string]: Resolution
+}
+
+export function addToShrinkwrap(
+  shrinkwrap: Shrinkwrap,
+  spec: PackageSpec,
+  resolution: Resolution
+): void {
+  switch (resolution.type) {
+    case 'package':
+      resolution = {...resolution}
+      delete resolution.pkg
+      shrinkwrap[spec.raw] = resolution
+      break;
+    case 'tarball':
+    case 'git-repo':
+      shrinkwrap[spec.raw] = {...resolution}
+      break;
+  }
```

Comment:
I think we should add `excludeFromShrinkwrap: boolean` attribute to `Resolution` which controls if any particular resolution should participate in shrinkwrapping or not. Thoughts? `pkg` should be stripped for each resolution as it is just for caching purposes.

This will allow custom resolvers to store any info in shrinkwrap as needed.